File: src/main/scala/demo/secondSortDemo/demo2.scala
package demo.secondSortDemo

import org.apache.spark.{SparkConf, SparkContext}

/**
  * Created by wanglei on 2018/1/1.
  * 二次排序
  *
  * 案例：
  * 1、按照文件中的第一列排序
  * 2、如果第一列相同，则按照第二列排序
  */
object demo2 {

  def main(args: Array[String]): Unit = {
    val sparkConf = new SparkConf().setAppName(this.getClass.getName.replace("$", "")).setMaster("local[1]")
    val sparkContext = new SparkContext(sparkConf)
    val rdd = sparkContext.textFile("F:\\IdeaProjects\\MySpark\\src\\testData\\sort.txt")
    val pairs = rdd.map {
      line => (new SecondSort(line.split(" ")(0).toInt, line.split(" ")(1).toInt), line)
    }
    val sortedPairs = pairs.sortByKey()
    val sortLines = sortedPairs.map(sortedPair => sortedPair._2)
    sortLines.foreach(println(_))
  }

}

File: src/main/java/PlayCode/Demo/打印/Demo2.java
package PlayCode.Demo.打印;
/*
1+2+3+4+5+6+7+8+9+10=55
21+22+23+24+25+26+27+28+29+30=255
....
90+91+..............+100=955
*/

class Demo2 {
	public static void main(String[] args) {
		   
	     for(int i=1;i<=100;i+=10){
			   int sum=0;
			 for(int j=i;j<=(i+9);j++){
				if(j!=(i+9)){
					System.out.print(j+"+");
					sum+=j;
				}
				
				else{
					sum = sum + j;
					System.out.print(j+"="+sum);
				}
				 
			 }System.out.println();
	     }
	}
}
